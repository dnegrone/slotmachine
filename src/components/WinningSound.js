import React from 'react';

const WinningSound = () => {
    
    return (
        <audio autoplay="autoplay" className="player" preload="false">
          <source src="./assets/winning_slot.wav" />
        </audio>  
    );
    
}

export default WinningSound;