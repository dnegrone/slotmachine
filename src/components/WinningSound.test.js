import React from 'react';
import { shallow } from 'enzyme';
import WinningSound from './WinningSound';

describe('Component Winning Sound <WinningSound />', () => {
    const wrapper = shallow(<WinningSound />);
    const mockCallBack = jest.fn();
    
    it('expects Winning Sound component to be rendered', () => {
        expect(wrapper.debug()).toMatchSnapshot();
    })
    
    it('expect to have a player', () => {
        // console.log(wrapper.debug());
        expect(wrapper.find('.player').length).toEqual(1);
    })

    it('expect to autoplay without preloading a sound', () => {
        expect(wrapper.find({ autoplay: 'autoplay' }));
        expect(wrapper.find({ preload: false }));
    })
    
    it('expect to have a sound', () => {
        expect(wrapper.find('source').length).toEqual(1);
    })

    it('expect the sound to be winning_slot.wav', () => {
        expect(wrapper.find({ src: 'winning_slot.wav' }));
    })

})