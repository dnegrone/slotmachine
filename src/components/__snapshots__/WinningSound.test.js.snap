// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Winning Sound <WinningSound /> expects Winning Sound component to be rendered 1`] = `
"<audio autoplay=\\"autoplay\\" className=\\"player\\" preload=\\"false\\">
  <source src=\\"./assets/winning_slot.wav\\" />
</audio>"
`;
